sentry_sdk/integrations/celery.py

```diff
@@ -387,123 +392,47 @@ def _get_monitor_config(celery_schedule, app):
     return monitor_config
 
 
-def _reinstall_patched_tasks(app, sender, add_updated_periodic_tasks):
-    # type: (Celery, Service, List[functools.partial[Any]]) -> None
-
-    # Stop Celery Beat
-    sender.stop()
-
-    # Update tasks to include Monitor information in headers
-    for add_updated_periodic_task in add_updated_periodic_tasks:
-        add_updated_periodic_task()
-
-    # Start Celery Beat (with new (cloned) schedule, because old one is still in use)
-    cloned_schedule = tempfile.NamedTemporaryFile(suffix="-patched-by-sentry-sdk")
-    with open(sender.schedule_filename, "rb") as original_schedule:
-        shutil.copyfileobj(original_schedule, cloned_schedule)
+def _patch_beat_apply_entry():
+    # type: () -> None
+    original_apply_entry = Scheduler.apply_entry
+
+    def sentry_apply_entry(*args, **kwargs):
+        # type: (*Any, **Any) -> None
+        scheduler, schedule_entry = args
+        app = scheduler.app
+
+        celery_schedule = schedule_entry.schedule
+        monitor_config = _get_monitor_config(celery_schedule, app)
+        monitor_name = schedule_entry.name
+
+        headers = schedule_entry.options.pop("headers", {})
+        headers.update(
+            {
+                "sentry-monitor-slug": monitor_name,
+                "sentry-monitor-config": monitor_config,
+            }
+        )
 
-    app.Beat(schedule=cloned_schedule.name).run()
+        check_in_id = capture_checkin(
+            monitor_slug=monitor_name,
+            monitor_config=monitor_config,
+            status=MonitorStatus.IN_PROGRESS,
+        )
+        headers.update({"sentry-monitor-check-in-id": check_in_id})
 
+        schedule_entry.options.update(headers)
+        original_apply_entry(*args, **kwargs)
```

Comment:
do you have to return the value to the original function?

Reply:
good catch. The original one does not have an return value, but it can not hurt. Makes it forward compatible